flict/flictlib/project/reader.py

```diff
@@ -140,23 +146,30 @@ def read_project(self, project_file):
             "packages": flat_packages,
         }
 
+    def _relationship_is_dependency(self, relationshiptype):
+        return relationshiptype in DEPENDENCY_TAGS
+
     def _read_spdx_2_2(self, only_packages=None):
         project_name = self.project['name']
         packages = {}
         for pkg in self.project['packages']:
             elem_id = pkg['SPDXID']
             packages[elem_id] = {
                 'id': elem_id,
-                'name': pkg['name'],
-                'version': pkg['versionInfo'],
-                'license': pkg['licenseConcluded'],
-                'description': pkg['description'],
+                'name': pkg.get('name', ""),
+                'version': pkg.get('versionInfo', ""),
+                'license': pkg.get('licenseConcluded'),
+                'description': pkg.get('description'),
                 'dependencies': [],
             }
 
-        if 'relationships' in self.project:
-            for dep in self.project['relationships']:
-                top_package = dep['relatedSpdxElement']
+        for dep in self.project.get('relationships', []):
+            top_package = dep['relatedSpdxElement']
+            relationshiptype = dep['relationshipType']
+            if not self._relationship_is_dependency(relationshiptype):
+                logging.debug(f"relationship {dep.get('spdxElementId')} ignored since {relationshiptype} is not defined to be a dependency tag.")
```

Comment:
AFAIS this code path is missing a test case - please add one

Reply:
`    def test_zlib_with_deps(self)` in `tests/test_verify.py`does this

Reply:
Added test case as a response to your comment. coming with the push